### DISCUSSION TITLE
Does brew support using a non-admin account?

### DISCUSSION THREAD

**Discussion by GAM-GC** [Homebrew/discussions#4149] [Everyday usage]
### Output of `brew config`

```shell
bash-3.2$ brew config
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 9 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 482f27271966a2712680195430ca58c37717c765
Core tap last commit: 14 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.38.1 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.2-x86_64
CLT: N/A
Xcode: 14.2
bash-3.2$
```

### Output of `brew doctor`

```shell
bash-3.2$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  web100clt

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libmp3lame.0.0.0.dylib
  /usr/local/lib/libssl.0.9.8.dylib
  /usr/local/lib/libwpsapi.2.0.7.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/lame/lame.h

Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.la' files:
  /usr/local/lib/libltdl.la
  /usr/local/lib/libmp3lame.la
  /usr/local/lib/libwpsapi.la

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libmp3lame.a
  /usr/local/lib/libwpsapi.a
bash-3.2$
```

### Description of issue

Hi,

An important security protocol in our org is that everyone, including IT, uses non-admin accounts for day to day use on their machines - Mac & PC. Because of this, I use brew in the admin account on my machine (as it seemed to require that when I first started to use it) and the software it controls (e.g. MySQL) in my non-admin account. After experiencing an issue with brew after su-ing to my admin account I came across [this thread](https://github.com/Homebrew/brew/issues/6439) where I got the distinct impression from @MikeMcQuaid that this was not supported.

Quite apart from the counter-intuitive requirement to be running a gui with a command-line tool, what is the brew account’s requirements in terms of it being an admin? Does it insist that the account be such? How does that square with the fact that very many orgs, as ours, view using an admin account for day to day use (such as for brew) as a security risk?  If what I’m (trying to) do isn’t supported, (how) can I properly use brew in my non-admin account; i.e. without hacking stuff?

Thanks,
GAM

**Comment by SMillerDev** [comment]
Using Homebrew in a multi user environment is not something that it's optimised for so you'll have to find your own solutions to any problems with it.

Homebrew works just fine without admin access though. It only needs admin rights for the initial install. It will even refuse to work as root.

**Reply by GAM-GC** [reply]
Thanks.
So the answer would be, install using admin account and then just use in non-admin account?

**Reply by SMillerDev** [reply]
You'll need to fix the permissions on the files after the install, but yes.

**Reply by averms** [reply]
> You'll need to fix the permissions on the files after the install, but yes.

What are the fixes? I'm guessing `chown` something?

**Reply by Jerry1144** [reply]
@averms Just `brew doctor` with the non-admin account and Homebrew will present these:
```
Warning: The following directories are not writable by your user:
/opt/homebrew
/opt/homebrew/Cellar
/opt/homebrew/Frameworks
/opt/homebrew/bin
/opt/homebrew/etc
/opt/homebrew/etc/bash_completion.d
/opt/homebrew/include
/opt/homebrew/lib
/opt/homebrew/lib/pkgconfig
/opt/homebrew/opt
/opt/homebrew/sbin
/opt/homebrew/share
/opt/homebrew/share/aclocal
/opt/homebrew/share/doc
/opt/homebrew/share/info
/opt/homebrew/share/locale
/opt/homebrew/share/man
/opt/homebrew/share/man/man1
/opt/homebrew/share/man/man3
/opt/homebrew/share/man/man5
/opt/homebrew/share/man/man7
/opt/homebrew/share/man/man8
/opt/homebrew/share/zsh
/opt/homebrew/share/zsh/site-functions
/opt/homebrew/var/homebrew/linked
/opt/homebrew/var/homebrew/locks

You should change the ownership of these directories to your user.
  sudo chown -R Jerry1144 /opt/homebrew /opt/homebrew/Cellar /opt/homebrew/Frameworks /opt/homebrew/bin /opt/homebrew/etc /opt/homebrew/etc/bash_completion.d /opt/homebrew/include /opt/homebrew/lib /opt/homebrew/lib/pkgconfig /opt/homebrew/opt /opt/homebrew/sbin /opt/homebrew/share /opt/homebrew/share/aclocal /opt/homebrew/share/doc /opt/homebrew/share/info /opt/homebrew/share/locale /opt/homebrew/share/man /opt/homebrew/share/man/man1 /opt/homebrew/share/man/man3 /opt/homebrew/share/man/man5 /opt/homebrew/share/man/man7 /opt/homebrew/share/man/man8 /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions /opt/homebrew/var/homebrew/linked /opt/homebrew/var/homebrew/locks

And make sure that your user has write permission.
  chmod u+w /opt/homebrew /opt/homebrew/Cellar /opt/homebrew/Frameworks /opt/homebrew/bin /opt/homebrew/etc /opt/homebrew/etc/bash_completion.d /opt/homebrew/include /opt/homebrew/lib /opt/homebrew/lib/pkgconfig /opt/homebrew/opt /opt/homebrew/sbin /opt/homebrew/share /opt/homebrew/share/aclocal /opt/homebrew/share/doc /opt/homebrew/share/info /opt/homebrew/share/locale /opt/homebrew/share/man /opt/homebrew/share/man/man1 /opt/homebrew/share/man/man3 /opt/homebrew/share/man/man5 /opt/homebrew/share/man/man7 /opt/homebrew/share/man/man8 /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions /opt/homebrew/var/homebrew/linked /opt/homebrew/var/homebrew/locks
```

You'll need an admin account to run the `sudo chown` part though. After that things should mostly workout.

**Comment by Jerry1144** [comment]
> the counter-intuitive requirement to be running a gui with a command-line tool

The requirement is just particular to how `brew services` implements launching a LaunchAgent. Non-sudo invocation of `brew services start` will register that service at ~/Library/LaunchAgents, and then try to start it. Since you ran that with your headless admin account, it got registered with that admin account, but nowhere else. Then launchctl fails to start that service for `domain gui/501` non existent blah, but the real problem is it was registered with an infrequently-accessed account, so won't get run that much (definitely not every restart) even if the invocation succeeded.

You really want to do it using your non-admin, everyday account, because every user has total control of their own ~/Library/LaunchAgents. Or just `sudo brew services start` which registers a Launch Daemon that serves every user within one process.
